components/screens/ChatScreen.js

```diff
@@ -10,6 +10,22 @@ import ConfirmDeleteMessageButton from '../buttons/ConfirmDeleteMessageButton';
 import RejectDeleteMessageButton from '../buttons/RejectDeleteMessageButton';
 
 const MessageItem = ({ item, sender, setDeletdedMessage, setModalVisible, setSentAt }) => {
+    const [message, setMessage] = useState(null)
+    const [error, setError] = useState(null)
+
+    const handleDecryption = () => {
+        try {
+            setMessage(decryptMessage(item.message))
+        } catch (err) {
+            console.log(err)
+            setError(err)
+        }
+    }
+
+    useEffect(() => {
+        handleDecryption()
+    }, [])
+
     if (sender.id === auth.currentUser.uid) {
         const date = new Date(item.sentAt)
         const formattedDate = date.getHours() + ':' + date.getMinutes()
```

Comment:
This if/else blocks seems very redundant! 
Try to do the redundant work above the if and only render the differed components inside the if/else.